MaxText/max_utils.py

```diff
@@ -362,6 +362,21 @@ def fill_unspecified_mesh_axes(parallelism_vals, target_product, parallelism_typ
   return parallelism_vals
 
 
+def reshape_mesh_to_rings(a):
```

Comment:
Thanks Raymond! I think once you merged the code, I will modify this function a little bit to be general to fit in 32x8 split ([this](b/379317193) for your reference).